### PR TITLE
chore: PPO system cleanup

### PR DESCRIPTION
**Pull request by sash-a**
## What?
Did a couple small clean ups to the PPO systems that are long overdue:
* Removed all the capital comments
* Removed opt states from the losses
* Removed tree map around the done reshape during acting (it's just an array)
* Removed metrics from the traj batch because they aren't really part of the batch
* Cleaner unpacking of the loss metrics

@RuanJohn should we add this for MAT?

### CODE REVIEW

File: mava/systems/mat/anakin/mat.py [hunk 1]
```diff
@@ -83,51 +84,35 @@ def _update_step(learner_state: LearnerState, _: Any) -> Tuple[LearnerState, Tup
...
-            )
+            done = timestep.last().repeat(env.num_agents).reshape(config.arch.num_envs, -1)
             transition = PPOTransition(
-                done,
-                action,
-                value,
-                timestep.reward,
-                log_prob,
-                last_timestep.observation,
-                info,
```
Comment (RuanJohn) [inline mava/systems/mat/anakin/mat.py:120]: A slight concern I have with not storing info in the transition is that we lose the ability to quickly edit systems that could need it in the future. What do you think?

File: mava/systems/mat/anakin/mat.py [hunk 2]
```diff
@@ -208,28 +190,27 @@ def _loss_fn(
                         )
                         * gae
                     )
-                    loss_actor = -jnp.minimum(loss_actor1, loss_actor2)
-                    loss_actor = loss_actor.mean()
+                    actor_loss = -jnp.minimum(loss_actor1, loss_actor2)
```
Comment (RuanJohn) [inline mava/systems/mat/anakin/mat.py]: If we change from `loss_actor` -> `actor_loss` can we please do the same for `loss_actor1` and `loss_actor2`

File: mava/systems/mat/anakin/mat.py [hunk 3]
```diff
@@ -248,15 +229,11 @@ def _loss_fn(
...
-                # UPDATE ACTOR PARAMS AND OPTIMISER STATE
+                # Update params and optimiser state
                 actor_updates, new_opt_state = actor_update_fn(actor_grads, opt_state)
                 new_params = optax.apply_updates(params, actor_updates)
 
-                # PACK LOSS INFO
-                total_loss = actor_loss_info[0]
-                value_loss = actor_loss_info[1][2]
-                actor_loss = actor_loss_info[1][0]
-                entropy = actor_loss_info[1][1]
```
Comment (RuanJohn) [inline mava/systems/mat/anakin/mat.py:259]: Why do we no longer need to use this?

File: mava/systems/ppo/anakin/ff_ippo.py [hunk 4]
```diff
@@ -253,39 +232,36 @@ def _critic_loss_fn(
...
-                entropy = actor_loss_info[1][1]
+                actor_loss, (raw_actor_loss, entropy) = actor_loss_info
+                critic_loss, raw_critic_loss = critic_loss_info
+
+                total_loss = actor_loss + critic_loss
                 loss_info = {
                     "total_loss": total_loss,
-                    "value_loss": value_loss,
-                    "actor_loss": actor_loss,
+                    "value_loss": raw_critic_loss,
```
Comment (RuanJohn) [inline mava/systems/ppo/anakin/ff_ippo.py]: If we change value -> critic I think we should be consistent with the naming everywhere. Are you keeping it as value to still be comparable with old experiments?

File: mava/systems/ppo/anakin/ff_ippo.py [hunk 5]
```diff
@@ -253,39 +232,36 @@ def _critic_loss_fn(
...
-                # PACK NEW PARAMS AND OPTIMISER STATE
                 new_params = Params(actor_new_params, critic_new_params)
                 new_opt_state = OptStates(actor_new_opt_state, critic_new_opt_state)
 
-                # PACK LOSS INFO
-                total_loss = actor_loss_info[0] + critic_loss_info[0]
-                value_loss = critic_loss_info[1]
-                actor_loss = actor_loss_info[1][0]
-                entropy = actor_loss_info[1][1]
+                actor_loss, (raw_actor_loss, entropy) = actor_loss_info
```
Comment (RuanJohn) [inline mava/systems/ppo/anakin/ff_ippo.py]: No so sure about appending `raw_` to the loss values. I think `actor_loss` and `raw_actor_loss` are the same in this case as well. So we could always just catch one of them with a `_` and used the other.

File: mava/systems/ppo/anakin/ff_mappo.py [hunk 6]
```diff
@@ -74,39 +73,36 @@ def _update_step(learner_state: LearnerState, _: Any) -> Tuple[LearnerState, Tup
...
+            # Step environment
             env_state, timestep = jax.vmap(env.step, in_axes=(0, 0))(env_state, action)
 
-            # LOG EPISODE METRICS
-            done = tree.map(
-                lambda x: jnp.repeat(x, config.system.num_agents).reshape(config.arch.num_envs, -1),
-                timestep.last(),
-            )
-            info = timestep.extras["episode_metrics"]
+            done = timestep.last().repeat(env.num_agents).reshape(config.arch.num_envs, -1)
```
Comment (RuanJohn) [inline mava/systems/ppo/anakin/ff_mappo.py:92]: I like this!

File: mava/systems/ppo/anakin/ff_mappo.py [hunk 7]
```diff
@@ -144,23 +140,21 @@ def _update_epoch(update_state: Tuple, _: Any) -> Tuple:
 
             def _update_minibatch(train_state: Tuple, batch_info: Tuple) -> Tuple:
                 """Update the network for a single minibatch."""
-                # UNPACK TRAIN STATE AND BATCH INFO
                 params, opt_states, key = train_state
                 traj_batch, advantages, targets = batch_info
 
                 def _actor_loss_fn(
                     actor_params: FrozenDict,
-                    actor_opt_state: OptState,
```
Comment (RuanJohn) [inline mava/systems/ppo/anakin/ff_mappo.py:153]: Long overdue!

**Review by RuanJohn — CHANGES_REQUESTED** [review]
Thank you @sash-a! I think these are some very needed changes. I just have a few suggestions and questions. Any comment applies to all systems.

**Comment by sash-a** [comment]
todo: same fixes for sable and switch sable to using the PPOTransition

File: README.md [hunk 8]
```diff
@@ -184,10 +184,6 @@ python mava/systems/ff_ippo.py env=rware env/scenario=tiny-4ag
 
 Additionally, we also have a [Quickstart notebook][quickstart] that can be used to quickly create and train your first Multi-agent system.
 
-## Advanced Usage 👽
```
Comment (WiemKhlifi) [inline README.md:187]: Are we giving up on this because it's no longer maintained?

Comment (sash-a) [reply README.md:187]: Ya @RuanJohn and I discussed it. I think if we want to bring it back at some point it would be nice, but right now it's not maintained and I'm not even sure if it works right now

Comment (WiemKhlifi) [reply README.md:187]: Indeed I agree on that it has some issues within the evaluator as well, hpefully we can restore it in future 🤞

**Review by OmaymaMahjoub — DISMISSED** [review]
LGTM! Let's just wait for the benchmark before we merge this 🙏

**Review by RuanJohn — DISMISSED** [review]
:trophy:

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @sash-a 🙌

**Review by RuanJohn — APPROVED** [review]
:star2:
